### DISCUSSION TITLE
How might multilingual work here?

### DISCUSSION THREAD

**Discussion by bdarcus** [bdarcus/csl-next-attic#6] [Q&A]
ATM, I have no idea, but it's likely not worth doing at all if we can't figure out how to do it elegantly.

**Comment by denismaier** [comment]
I think there are basically two options.

1. Use some sort of preorocessing hook. Before a citeproc sees the multilingual data it has already been assembled into a string. I think that's basically how jurism works.
2. Let templates do all the heavy lifting. That is, some styles will ignore the multilingual data, in others will make heavy use if it.
3. Non-multilingual styles could possibly made multilingual by patching specific templates.

**Reply by bdarcus** [reply]
It seems either 1 or 2 makes most sense, and someone with knowledge of the area will have to test it with demo templates?

But it sounds like you're optimistic, so maybe we could defer that?

**Reply by denismaier** [reply]
Yes, I'm somewhat optimistic. I guess it would still be good to have input from @fbennett here.

**Reply by fbennett** [reply]
> Use some sort of preorocessing hook. Before a citeproc sees the multilingual data it has already been assembled into a string. I think that's basically how jurism works.

In Jurism, multilingual inputs are tagged with a language variant at the level of individual fields, and a map generated from the user's configuration settings controls which variants of a field are rendered in the primary position defined by the CSL style, and which may follow it, set off by a delimiter specified again in user settings.

This works against all standard CSL styles without modification, but it's less a form of pre-processing and more a deep integration of multilingual support in the processor implementation.

**Reply by bdarcus** [reply]
Thanks @fbennett! Do you think that could generalize to CSL more widely?

Like, what do you mean by:

>  ... it's less a form of pre-processing and more a deep integration of multilingual support in the processor implementation.

**Comment by bdarcus** [comment]
Someone the typst trackers says this is current state of the work in general; not sure if it has any relevance here.

https://projectfluent.org/

And when done:

https://messageformat.github.io/messageformat/v2/

Actually, may be more relevant to terms, or not at all.

**Comment by fbennett** [comment]
I'm not sure whether the behavior could be specified without wading into
implementation details. The Jurism language configuration menus have a
two-tier structure similar to the role/permission settings found in a CMS.
Other projects might want to adopt a different control mechanism, so it
might be excessively constraining to specify the shape of language
configuration data sent to the processor.

**Reply by denismaier** [reply]
Could this configuration be sent separately to the processor. Like when starting the processor? Obviously, this should be optional.

**Comment by fbennett** [comment]
Yes, if I recall correctly, the configuration is supplied when the processor is instantiated, and the processor provides a hook that the calling application can use to adjust the configuration without tearing the processor down and rebuilding it. I guess this is a good example of how the "injection" approach tends to draw discussion away from broad standards and toward implementation details.
